deviceshifu/pkg/deviceshifu/deviceshifu.go

```diff
@@ -180,11 +191,35 @@ func (handler DeviceCommandHandlerHTTP) commandHandleFunc() http.HandlerFunc {
 			}
 		}
 
-		log.Printf("handling instruction '%v' to '%v'", handlerInstruction, *handlerEdgeDeviceSpec.Address)
-		resp, err := handlerHTTPClient.Get("http://" + *handlerEdgeDeviceSpec.Address + "/" + handlerInstruction)
-		if err != nil {
-			http.Error(w, err.Error(), http.StatusServiceUnavailable)
-			log.Printf("HTTP error" + err.Error())
+		var resp *http.Response
+		var httpErr error
+		reqType := r.Method
+
+		log.Printf("handling instruction '%v' to '%v' with request type %v", handlerInstruction, *handlerEdgeDeviceSpec.Address, reqType)
+
+		if reqType == "GET" {
+			resp, httpErr = handlerHTTPClient.Get("http://" + *handlerEdgeDeviceSpec.Address + "/" + handlerInstruction)
+			if httpErr != nil {
+				http.Error(w, httpErr.Error(), http.StatusServiceUnavailable)
+				log.Printf("HTTP GET error" + httpErr.Error())
+			}
+		} else if reqType == "POST" {
+			requestBody, parseErr := io.ReadAll(r.Body)
+			if parseErr != nil {
+				log.Panic("Error on parsing body" + parseErr.Error())
+				return
+			}
+			queryStr := createQueryStringFromRequest(r)
+			resp, httpErr = handlerHTTPClient.Post("http://"+*handlerEdgeDeviceSpec.Address+"/"+handlerInstruction+"?"+queryStr,
+				"application/json", bytes.NewBuffer(requestBody))
```

Comment:
instead of specifying `"application/json"`, can we just proxy whatever is from the original request?